mlir/utils/jenkins/parameterSweeps.py

```diff
@@ -0,0 +1,394 @@
+    os.close(runnerFromLowering)
+
+    _, loweringErrs = await lowering.communicate(input=highLevel)
+    runnerOut, runnerErrs = await runner.communicate()
+    runnerOut = runnerOut.decode('utf-8')
+
+    if lowering.returncode != 0:
+        if DEBUG:
+            print(f"""Low-level lowering did not complete succesfully for config {config!r}
+Command line = {miopenGenOpts}
+Errors = {loweringErrs.decode('utf-8')}
+Return code = {lowering.returncode}""")
+        return TestResult.FAIL
+
+    if runner.returncode != 0:
+        if DEBUG:
+            print(f"""Runner execution failed for config {config!r}
+Output = {runnerOut}
+Errors = {runnerErrs.decode('utf-8')}
+Return code = {runner.returncode}""", file=sys.stderr)
+        return False
+
+    if not CORRECT_RESULT_RE.search(runnerOut):
+        print(f"""Convolution returned intorrect result
+Output = {runnerOut}
+Errors = {runnerErrs.decode('utf-8')}""", file=sys.stderr)
+        return TestResult.FAIL
+    return TestResult.PASS
+
+def grouper(iterable, n):
+    it = iter(iterable)
+    while True:
+        chunk = tuple(itertools.islice(it, n))
+        if not chunk:
+            return
+        yield chunk
+
+async def dropGoodConfig(config: ConvConfiguration) -> Union[TestResult, ConvConfiguration]:
+    """Test the given `params`, returning the corresponding `config` on failure
+    and `None` on success or inapplicability"""
+    result = await testConfig(config)
+    if not QUIET:
+        print(f"{result.name}: {config!r}")
+    if result == TestResult.FAIL:
+        return config
+    return result
+
+IterType = TypeVar('IterType')
+async def sweepParameters(paramIter: Iterable[IterType],
+    toConfig: Callable[[IterType], MLIROnlyConfig]) -> Tuple[int, int, List[MLIROnlyConfig]]:
+
+    CONCURRENT_TESTS = 100
+    failingConfigs = []
+    passed = 0
+    invalid = 0
+    configs = (c for c in (toConfig(p) for p in paramIter))
+    for configs in grouper((dropGoodConfig(c) for c in configs), CONCURRENT_TESTS):
+        configsFuture = asyncio.gather(*configs)
+        try:
+            configsResults = await configsFuture
+        except Exception as e:
+            configsFuture.cancel()
+            raise e
+        for result in configsResults:
+            if result == TestResult.PASS:
+                passed = passed + 1
+            elif result == TestResult.INVALID:
+                invalid = invalid + 1
+            else:
+                failingConfigs.append(result)
+
+    return (passed, invalid, failingConfigs)
+
+CONV_STRUCTURE = itertools.product(
+    # Small/large
+    [False, True],
+    # op
+    ['fwd', 'wrw', 'bwd'],
+    # layout
+    ['NCHW', 'NHWC'],
+    # dtype
+    ['f32', 'f16'],
+    # Padding - hl, hr, wl, wr in [0, 3]
+    range(0, 4),
+    range(0, 4),
+    range(0, 4),
+    range(0, 4),
+    # Stride - 1 or 2
+    range(1, 3),
+    range(1, 3),
+    # Dilation in [1, 2]
+    range(1, 3),
+    range(1, 3))
```

Comment:
Some fairly important constants here: What's the reasoning in choosing the ranges? This question applies to other Structure configurations.

To make sure I understand: we brute force and try all and every config in the range? I'm asking because I was assuming that we "randomly pick" parameters.

Reply:
Yeah, no, this is very much a brute force check, hence running weekly

Reply:
(and I added some comments to clarify where I got the numbers from)